todo-mvc/src/actions/widgetTodoActions.ts

```diff
@@ -56,8 +55,9 @@ export const putTodo: AnyAction = createAction({
 			const item = puts[0];
 			const children = beforeAll.map((child: any) => child.id);
 
+			item.type = 'todo-item';
+
 			const put = function() {
-				app.registerWidgetFactory(item.id, createTodoItem);
 				return widgetStore
 				.put(item)
```

Comment:
Do you mean `Object.assign({}, item, { type: 'todo-item' });`? Because as I understand it `Object.assign` will modify the target object.